File: ts/lcof/53.ts
function missingNumber(nums: number[]): number {
  let l = 0, r = nums.length - 1, m = 0
  while (l <= r) {
    m = l + ((r - l) >> 1)
    if (m === nums[m]) l = m + 1
    else r = m - 1
  }
  return l
};

File: py3/lcof/53.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            m = l + (r - l) // 2
            if nums[m] == m:
                l = m + 1
            else:
                r = m - 1

        return l